apps/search/js/newtab.js

```diff
@@ -1,8 +1,17 @@
 'use strict';
-/* global asyncStorage */
+/* global asyncStorage, SettingsListener */
 
 (function(exports) {
 
+  SettingsListener.observe('browser.private.default', false, function (value) {
+    if (value) {
+      exports.newtab.pretendToBePrivate();
+    }
+    else {
+      exports.newtab.stopPretendingToBePrivate();
+    }
+  });
+
```

Comment:
I know you mentioned not using an observer here, but this actually works well. If I change the setting while the app is open it will toggle the expected functionality.